services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_dask_scheduler.py

```diff
@@ -209,24 +245,44 @@ async def _process_completed_tasks(
         user_id: UserID,
         tasks: list[CompTaskAtDB],
         iteration: Iteration,
-        pipeline_params: ScheduledPipelineParams,
+        comp_run: CompRunsAtDB,
     ) -> None:
         try:
-            async with _cluster_dask_client(user_id, pipeline_params, self) as client:
+            async with _cluster_dask_client(
+                user_id,
+                self,
+                use_on_demand_clusters=comp_run.use_on_demand_clusters,
+                cluster_id=(
+                    comp_run.cluster_id
+                    if comp_run.cluster_id is not None
+                    else DEFAULT_CLUSTER_ID
+                ),
+                run_metadata=comp_run.metadata,
+            ) as client:
                 tasks_results = await asyncio.gather(
                     *[client.get_task_result(t.job_id or "undefined") for t in tasks],
                     return_exceptions=True,
                 )
             await asyncio.gather(
                 *[
                     self._process_task_result(
-                        task, result, pipeline_params.run_metadata, iteration
+                        task, result, comp_run.metadata, iteration
                     )
                     for task, result in zip(tasks, tasks_results, strict=True)
                 ]
             )
         finally:
-            async with _cluster_dask_client(user_id, pipeline_params, self) as client:
+            async with _cluster_dask_client(
+                user_id,
+                self,
```

Comment:
MINOR: passing `self` as positional arguments is not very informative. Should we use here keyword argument to identify that `self` means to the called function?

Reply:
it's a private method. but will think about it.